### PR TITLE
Perform basic architecture check in the starter

### PR DESCRIPTION
**Pull request by gvallee**
**Description of the Pull Request (PR):**

Perform basic architecture check in the starter.

**This fixes or addresses the following GitHub issues:**

- Addresses #1488. 

**Before submitting a PR, make sure you have done the following:**

- Read the [Guidelines for Contributing](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](https://github.com/sylabs/singularity/blob/master/CHANGELOG.md) if necessary according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added tests to validate this PR and tested this PR locally with a `make testall`
- Based this PR against the appropriate branch according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](https://github.com/sylabs/singularity/blob/master/CONTRIBUTORS.md)

Attn: @singularity-maintainers

### CODE REVIEW

**Comment by gvallee** [comment]
If I understand all the code correctly, note that it does not resolve the entire problem since when building/pulling an image, the architecture of the SIF file is based on the architecture of the Go runtime. I believe (to be confirmed) that we want to support cases where the creation of a local SIF is based only on getting file, no architecture-dependent actions, on an architecture that is not the architecture of the image. For example, we may want to let users create a SIF targeting the PPC64LE architecture on a amd64 system when we know that no action that require the target architecture is required during the creation of the SIF.
At the moment, the SIF architecture will be amd64 (the Go runtime architecture when creating the image, not PPC64LE, even if we are getting a PPC64LE image. Then, it is impossible to execute the image on PPC64LE since the architecture is amd64.

All this needs to be confirmed by someone who has a better understanding of the big picture. We still want to merge this PR meanwhile, even if it is more restrictive than we intent, it is still better that the current situation where we are not checking the architecture at any point.

**Comment by cclerget** [comment]
Yes, the pull step may not require architecture check, to allow to pull a ppc64 image, sign it and upload/run it on dedicated cluster.

@gvallee Typically for pull/build process :

1. pull image from docker/library/shub (pull/build command)
2. extract image filesystem to temporary folder (build command)
3. if definition file contains "executable" sections, the architecture check could occur at this step, but it could also be delegated to runtime which would fail during execution. (build command)

File: pkg/image/sif.go [hunk 1]
```diff
@@ -38,6 +40,12 @@ func (f *sifFormat) initializer(img *Image, fileinfo os.FileInfo) error {
 		return err
 	}
 
+	// Check the compatibility of the image's target architecture
+	sifArch := string(fimg.Header.Arch[:sif.HdrArchLen-1])
+	if sifArch != sif.HdrArchUnknown && sifArch != sif.GetSIFArch(runtime.GOARCH) {
```
Comment (cclerget) [inline pkg/image/sif.go:51]: like you said it's good for now. Comment for reminder, architecture like i386 can run on amd64, maybe some arm processor can run <= armv6 instructions on aarch64, however not 100% sure.

**Review by cclerget — APPROVED** [review]

File: pkg/image/sif.go (hunk 1 above)
Comment (gvallee) [reply pkg/image/sif.go:51]: Good idea and done. I tried to capture the entire situation and hopefully will work on better support during the next few weeks. Meanwhile I believe this PR is ready to go.

**Pull request merged**